### PR TITLE
Helmet

### LINKED ISSUE #42: feat: support react-helmet
Currently, there's no way to properly modify the head section from within a component or page. `react-helmet` would be a natural choice here as they also support SSR https://www.npmjs.com/package/react-helmet#server-usage

### PR DESCRIPTION
**Pull request by hannessolo**
Fixes #42

### CODE REVIEW

File: test/test-project/package.json [hunk 1]
```diff
@@ -1,6 +1,8 @@
 {
   "dependencies": {
-    "franklin-esr": "adobe-rnd/franklin-esr"
+    "babel-plugin-styled-components": "^2.0.7",
```
Comment (icaraps) [inline test/test-project/package.json:3]: Is `babel-plugin-styled-components` and `react-helmet` required for the test project?

File: renderer/_default.page.server.jsx [hunk 2]
```diff
@@ -24,28 +22,31 @@ async function render(pageContext) {
...
   documentHtml = escapeInject`<!DOCTYPE html>
     <html lang="en">
       <head>
-        <title>${title}</title>
+        ${dangerouslySkipEscape(helmet.title.toString())}
+        ${dangerouslySkipEscape(helmet.link.toString())}
+        ${dangerouslySkipEscape(helmet.meta.toString())}
         <meta charset="utf-8">
         <meta name="viewport" content="width=device-width, initial-scale=1">
         <meta name="description" content="demo">
```
[the remark this answers is not in the document]
Comment (mhaack) [reply renderer/_default.page.server.jsx]: Yes lets move that to Helmet as well, not hard code it.

**Comment by icaraps** [comment]
Can you pls verify if it works too when doing client-side navigation ?

**Comment by icaraps** [comment]
Do you also have this warning when running `npm run dev` ?

Screen Shot 2022-06-10 at 9 28 00 PM

File: components/PageLayout.jsx [hunk 3]
```diff
@@ -1,9 +1,19 @@
...
 
 export default function PageLayout({ pageContext, children }) {
   return (
     <PageContextProvider pageContext={pageContext}>
+      <Helmet>
+        <meta charSet="utf-8" />
+        <meta name="viewport" content="width=device-width, initial-scale=1" />
+        <meta name="description" content="demo" />
+        <title>{getPageTitle(pageContext)}</title>
+        <link rel="stylesheet" href="/fonts.css" media="print" onLoad="this.media='all'" />
```
Comment (icaraps) [inline components/PageLayout.jsx:14]: This is a problem when doing client-side navigation, the font will hide and show on every page navigation.
Can we keep the fonts.css and index.css links in the default page server renderer (so not within Helmet) or do you think there's a problem if we do that ?

**Review by icaraps — CHANGES_REQUESTED** [review]

**Comment by hannessolo** [comment]
> Do you also have this warning when running `npm run dev` ? I don't have it in worker mode though ...
> 
> Screen Shot 2022-06-10 at 9 28 00 PM

Yes, looks like an open issue. https://github.com/nfl/react-helmet/issues/548

The fix seems to be to use this fork until they fix it: https://www.npmjs.com/package/react-helmet-async

File: package.json [hunk 4]
```diff
@@ -30,6 +30,7 @@
     "node-fetch": "^3.2.4",
     "react": "^18.0.0",
     "react-dom": "^18.0.0",
+    "react-helmet": "^6.1.0",
```
Comment (icaraps) [inline package.json:33]: Can you pls try out https://www.npmjs.com/package/react-helmet-async ?

Comment (hannessolo) [reply package.json:33]: Yep, I took a look. But it looks like it will have issues with streaming. https://github.com/staylor/react-helmet-async#streams

**Review by icaraps — APPROVED** [review]

**Pull request merged**